shared_model/interfaces/iroha_internal/query_response_factory.hpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_QUERY_RESPONSE_FACTORY_HPP
+#define IROHA_QUERY_RESPONSE_FACTORY_HPP
+
+#include <memory>
+
+#include "interfaces/query_responses/block_query_response.hpp"
+#include "interfaces/query_responses/query_response.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    /**
+     * Factory for building query responses
+     */
+    class QueryResponseFactory {
+     public:
+      virtual ~QueryResponseFactory() = default;
+
+      /**
+       * Create response for account asset query
+       * @param assets to be inserted into the response
+       * @return account asset response
+       */
+      virtual std::unique_ptr<AccountAssetResponse> createAccountAssetResponse(
+          const std::vector<
+              std::shared_ptr<shared_model::interface::AccountAsset>>
+              &assets) = 0;
+
+      /**
+       * Create response for account detail query
+       * @param account_detail to be inserted into the response
+       * @return account detail response
+       */
+      virtual std::unique_ptr<AccountDetailResponse>
+      createAccountDetailResponse(const types::DetailType &account_detail) = 0;
+
+      /**
+       * Create response for account query
+       * @param account to be inserted into the response
+       * @param roles to be inserted into the response
+       * @return account response
+       */
+      virtual std::unique_ptr<AccountResponse> createAccountResponse(
+          const Account &account, const std::vector<std::string> &roles) = 0;
+
+      /**
+       * Describes type of error to be placed inside the error query response
+       */
+      enum class ErrorQueryType {
```

Comment:
Is it possible to reuse this list from somewhere?

Reply:
Such enumeration exists also in the form of variant inside `ErroQueryResponse`, but it's impossible, I think, to reuse it